### PR TITLE
Add development mode features for amp-story

### PR DESCRIPTION
**Pull request by newmuis**
Each page will show icons with warnings/errors, as below:

screen shot 2017-10-10 at 12 29 43 pm
screen shot 2017-10-10 at 12 29 49 pm

These are not validation errors; these are additional runtime checks to ensure that developers follow best practices for creating performant stories.  The list is maintained in logging.js.

### CODE REVIEW

**Comment by choumx** [comment]
This is cool. 😎

File: extensions/amp-story/0.1/system-layer.js [hunk 1]
```diff
@@ -146,6 +181,55 @@ export class SystemLayer {
   /**
    * @private
    */
+  buildForDevelopmentMode_() {
```
Comment (alanorozco) [inline extensions/amp-story/0.1/system-layer.js]: Does this get DCE'd?

Comment (newmuis) [reply extensions/amp-story/0.1/system-layer.js]: Updated this part

**Comment by bpaduch** [comment]
@newmuis - to clarify, if they get a green checkmark, is that a warning or is it "Hey you are doing it right and you get a check"?  I found that confusing when I ran into it.  If it's the latter, I wouldn't expect anything as I'm doing it correctly.

**Comment by newmuis** [comment]
It's the latter.  We can certainly explore changing that to work the way you've described.

**Comment by bpaduch** [comment]
Logged bug for language of success message vs. icon usage.

**Pull request merged**